test/apps/iodemo/io_demo.cc

```diff
@@ -1882,8 +1887,12 @@ class DemoClient : public P2pDemoCommon {
 
             r->init(this, server_index, msg->sn, conn->id(), opts().validate,
                     iov, 0);
-            conn->recv_am_data((*iov)[0].buffer(), msg->data_size,
-                               (*iov)[0].memh(), data_desc, r);
+            if (!ucx_am_is_rndv(data_desc)) {
+                conn->recv_am_data(NULL, 0, NULL, data_desc, r);
+            } else {
```

Comment:
why needed?
IOV must be valid after `prepare_am_recv_data_iov()`: https://github.com/openucx/ucx/blob/325a608ad53423a758f6e6944feb067cecefe1b3/test/apps/iodemo/io_demo.cc#L1023

Reply:
ditto